### PR TITLE
Tpetra:  deprecate readySend and Ssend options in Distributor

### PR DESCRIPTION
**Pull request by kddevin**
@trilinos/tpetra 

## Motivation

Tpetra did not test readySend or Ssend options, nor did any stakeholders use them.  See #9852  
This PR deprecates the options.  #9799 

This PR also changes the behavior when Isend is requested but not feasible.
The code used to throw an error, saying Isend would not work.  Why do that?  Why not just default back to send in those cases?  This PR make send the default in such cases, instead of throwing a cryptic error. 

## Testing

Tested on ascicgpu030 with and without Tpetra deprecated code.

### CODE REVIEW

File: packages/tpetra/core/src/Tpetra_Details_DistributorActor.hpp
Comment (tasmith4) [reply packages/tpetra/core/src/Tpetra_Details_DistributorActor.hpp:271]: oh I see what you mean now, and yeah, the comment must actually have been referring to that throw we used to have if barrierBetween wasn't true during readySend.  I don't know how to feel about doBarrier.  I don't love it, because I don't love having a bunch of production code that's only used to test things.  If it is easy to delete without rearranging a bunch of tests, I'm all for it.  My only concern would be if that messes up the testing logic in ways we don't understand.

File: packages/tpetra/core/src/Tpetra_Details_DistributorActor.hpp
Comment (tasmith4) [reply packages/tpetra/core/src/Tpetra_Details_DistributorActor.hpp:362]: I think a print statement that's only active in debug makes sense in this case.

For Rsend/Ssend, there is some protection, because a DEPRECATED_CODE=OFF build will give a compiler warning if they try to use those enum values.  I wouldn't mind a warning that we are converting Rsend/Ssend to Send printed always (release or debug), but only if DEPRECATED_CODE=ON.

File: packages/tpetra/core/src/Tpetra_Details_DistributorActor.hpp
Comment (kddevin) [reply packages/tpetra/core/src/Tpetra_Details_DistributorActor.hpp:271]: Thanks, @tasmith4 .  I will remove the doBarrier parameter option.  It was likely more useful during initial development of Distributor, and less useful now that Distributor is stable code.

File: packages/tpetra/core/src/Tpetra_Details_DistributorActor.hpp
Comment (kddevin) [reply packages/tpetra/core/src/Tpetra_Details_DistributorActor.hpp:362]: I'll add the print statements.

**Comment by kddevin** [comment]
@tasmith4  I think this PR is ready for re-review.  See several commits above that address your comments.  The PR tester reports failures, but the cdash page does not show any failures.  I am assuming a PR tester problem.

File: packages/tpetra/core/src/Tpetra_Details_DistributorActor.hpp
Comment (tasmith4) [reply packages/tpetra/core/src/Tpetra_Details_DistributorActor.hpp:366]: I'd argue that your new printout is sufficiently descriptive that it's better to delete the comment since we don't really need it anymore.  But I won't hold up the PR over this.

**Review by tasmith4 — APPROVED** [review]
Thanks @kddevin!

**Comment by william76** [comment]
@kddevin FYI, looks like this PR passed its tests, but it could go stale if it's not merged soon. I would hit the button but I wasn't sure if you were waiting on something else before merging in since the automerge flag wasn't set.

**Comment by kddevin** [comment]
Thanks, @william76.

**Pull request merged**
